src/App.tsx

```diff
@@ -1,24 +1,129 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { deleteTodo as deleteTodoById, getTodos, postTodo } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { AppError } from './types/ApiError';
+import { filterTodos, FilterType } from './helpers/filterTodos';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 6950;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorType, setErrorType] = useState(AppError.None);
+  const [filterType, setFilterType] = useState(FilterType.None);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [
+    deletingAllCompletedTodos,
+    setDeletingAllCompletedTodos,
+  ] = useState(false);
+
+  const fetchTodos = async () => {
+    try {
+      const apiTodos = await getTodos(USER_ID);
+
+      setTodos(apiTodos);
+    } catch {
+      setHasError(true);
+      setErrorType(AppError.Get);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const visibleTodos = useMemo(
+    () => filterTodos(todos, filterType),
+    [filterType, todos],
+  );
+
+  const addTodo = async (title: string) => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    const newTodo = await postTodo(title, USER_ID);
+
+    setTempTodo(null);
+    setTodos(prev => [...prev, newTodo]);
+  };
+
+  const deleteTodoFromServer = async (todoId: number) => {
+    await deleteTodoById(todoId);
+
+    fetchTodos();
```

Comment:
on my opinion, its better to filter existing todos in your app, instead of make a new request to the server